src/TraceEvent/Ctf/CtfReader.cs

```diff
@@ -90,14 +90,19 @@ public IEnumerable<CtfEventHeader> EnumerateEventHeaders()
 
             CtfInteger pid = null;
             CtfInteger tid = null;
+            CtfInteger vpid = null;
+            CtfInteger vtid = null;
             CtfArray processName = null;
             string lastProcessName = "";
             int processLen = 0;
             CtfStruct eventContext = _streamDefinition.EventContext;
             if (eventContext != null)
             {
-                pid = (CtfInteger)eventContext.GetField("_vpid")?.Type;
-                tid = (CtfInteger)eventContext.GetField("_vtid")?.Type;
+                vpid = (CtfInteger)eventContext.GetField("_vpid")?.Type;
```

Comment:
At this point, do we know if this is a kernelspace or userspace event?  I'd like to be able to deterministically select between vpid and vtid or pid and tid depending upon the channel (kernel or user) rather than depending upon the existence of one or the other.  It is technically possible for usermode events to contain context records for vpid, vtid, pid, and tid at the same time.

Reply:
Good point! I assumed we only turn on vpid and vtid for user mode. I need to investigate if user/kernel mode is known at this point. Also my question would be, under the scenario that both vpid and pid exist, we should probably store both fields, and which one should we use to match pids in startup? I assume vpid only exists if we use a container. (Didn't find good documentation online...)

Reply:
Just checked, we do know the domain information. It exists in the metadata passed to the reader. So by default, we will use vpid/vtid for user mode and pid/tid for kernel mode.